src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,207 @@
+     * @param elevatorSpeedPercentage the percentage of speed that the elevator will move in
+     * @return the command
+     */
+    public Command getSetTargetArmPositionCommand(Rotation2d targetAngle, double targetElevatorPosition, double angleSpeedPercentage, double elevatorSpeedPercentage) {
+        return new DeferredCommand(
+                () -> getCurrentSetTargetArmStateCommand(targetAngle, targetElevatorPosition, angleSpeedPercentage, elevatorSpeedPercentage),
+                Set.of(this)
+        );
+    }
+
+    private Command getCurrentSetTargetArmStateCommand(Rotation2d angle, double elevatorPosition, double angleSpeedPercentage, double elevatorSpeedPercentage) {
+        if (isElevatorOpening(elevatorPosition)) {
+            return new SequentialCommandGroup(
+                    getSetTargetAngleCommand(angle, angleSpeedPercentage),
+                    getSetTargetElevatorPositionCommand(elevatorPosition, elevatorSpeedPercentage)
+            );
+        }
+        return new SequentialCommandGroup(
+                getSetTargetElevatorPositionCommand(elevatorPosition, elevatorSpeedPercentage),
+                getSetTargetAngleCommand(angle, angleSpeedPercentage)
+        );
+    }
+
+    private Command getSetTargetAngleCommand(Rotation2d targetAngle, double speedPercentage) {
+        return new FunctionalCommand(
+                () -> generateAngleMotorProfile(targetAngle, speedPercentage),
+                this::setTargetAngleFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    private Command getSetTargetElevatorPositionCommand(double targetElevatorPosition, double speedPercentage) {
+        return new FunctionalCommand(
+                () -> generateElevatorMotorProfile(targetElevatorPosition, speedPercentage),
+                this::setTargetElevatorPositionFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    private boolean isElevatorOpening(double targetElevatorPosition) {
+        return getElevatorPositionRevolutions() < targetElevatorPosition;
+    }
+
+    private void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            armIO.stopAngleMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleMotorProfileTime());
+        double voltage = calculateAngleMotorOutput(targetState);
+        armIO.setAnglePower(voltage);
+    }
+
+    private void setTargetElevatorPositionFromProfile() {
+        if (elevatorMotorProfile == null) {
+            armIO.stopElevatorMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorMotorProfileTime());
+        double voltage = calculateElevatorMotorOutput(targetState);
+        armIO.setElevatorPower(voltage);
```

Comment:
Have a function called of the io `setTargetElevatorPosition` that accepts a rotation2d and the ff and move the pid calculation to the io.